Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/TaskViewModel.kt

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package org.dolphinemu.dolphinemu.model
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.*
+
+class TaskViewModel : ViewModel() {
+    val result = MutableLiveData<Int>()
+    val isComplete = MutableLiveData<Boolean>()
+    var cancelled = false
+    var isRunning = MutableLiveData<Boolean>()
+
+    lateinit var resultGetter: () -> Unit
```

Comment:
From the perspective of the calling code, the name `resultGetter` doesn't really explain what this variable is for. Maybe there's something better it could be named? `taskFunction`, `runnable` or something like that?